x-vector/src/X/Data/Vector/Generic.hs

```diff
@@ -14,12 +22,19 @@ module X.Data.Vector.Generic (
   , mapMaybeM
   , imapMaybeM
 
+  -- * Modifying vectors
+
+  -- ** Transposition
+  , transpose
+
```

Comment:
Do you think that the outer vector always has to be a boxed vector? The specific versions could be `Boxed.Vector (Boxed.Vector a)`, `Boxed.Vector (Unboxed.Vector a)` and `Boxed.Vector (Storable.Vector a)` respectively